src/persist/src/indexed/runtime.rs

```diff
@@ -158,6 +154,63 @@ where
     Ok(client)
 }
 
+/// WIP
+pub fn start_read<L, B>(
```

Comment:
we _could_ make this and `RuntimeImpl` share code, but these two are the ones where I really feel like a little duplication makes it all much easier to follow. wdyt?

Reply:
agree. we can also live with the duplication for a while and then decide to consolidate later?

Reply:
nope. what my comment here is trying to get at is that DRYing them is one of those cases where it ends up being _less_ maintainable and I've intentionally kept them separate